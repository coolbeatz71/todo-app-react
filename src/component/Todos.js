import React, { Component } from 'react';
import PropTypes from 'prop-types';
import TodoItem from './TodoItem';

/**
 * Display each todos
 * @class Todos
 */
class Todos extends Component {
  /**
   * Render the component
   *
   * @returns {void}
   * @memberof Todos
   */
  render() {
    const { todos, markComplete, delTodo } = this.props;
    return todos.map(todo => (
      <TodoItem key={todo.id} todo={todo} markComplete={markComplete} delTodo={delTodo} />
    ));
  }
}

Todos.propTypes = {
  todos: PropTypes.array.isRequired,
  markComplete: PropTypes.func.isRequired,
  delTodo: PropTypes.func.isRequired,
};

export default Todos;
