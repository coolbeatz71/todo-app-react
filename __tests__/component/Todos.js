import { shallow, mount } from 'enzyme';
import React from 'react';
import Todos from '../../src/component/Todos';

let wrapper;
const todos = [
  {
    id: 1,
    userId: 1,
    title: 'Just for testing 1',
    completed: false,
  },
  {
    id: 2,
    userId: 1,
    title: 'Just for testing 2',
    completed: false,
  },
  {
    id: 3,
    userId: 1,
    title: 'Just for testing 3',
    completed: true,
  },
];
const props = {
  todos,
  markComplete: jest.fn(),
  delTodo: jest.fn(),
};

describe('Todos component', () => {
  beforeEach(() => {
    wrapper = mount(<Todos {...props} />);
  });

  it('Should render the Todos component', () => {
    wrapper = shallow(<Todos {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
