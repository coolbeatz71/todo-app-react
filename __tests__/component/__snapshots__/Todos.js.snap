// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todos component Should render the Todos component 1`] = `ShallowWrapper {}`;
